File: prid-2021-002/backend/models/Board.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Diagnostics;
using System.Linq;
using Microsoft.EntityFrameworkCore;

namespace prid_2021_002.Models {

    public class Board {
        [Key]
        public int BoardId { get; set; }

        [Required(ErrorMessage = "Required")]
        public string Title { get; set; }

        public DateTime Timestamp { get; set; } = DateTime.Now;

        public string PicturePath { get; set; }

        public virtual User Author { get; set;}
        public int AuthorId { get; set; }

        public virtual IList<List> Lists { get; set; } = new List<List>();

        public virtual IList<UserBoard> BoardUsers { get; set; } = new List<UserBoard>();

        [NotMapped]
        public IEnumerable<User> Collaborators {
            get => BoardUsers.Select(ub => ub.User);
        }
   
    }
}

File: prid-2021-002/frontend/src/app/core/models/board.ts
import { List } from "./list";
import { User } from "./User";

export class Board {
    boardId: number;
    title: string;
    timestamp: string;
    picturePath: string;
    author: User;
    authorId: number;
    collaborators: User[];
    lists: List[];

    constructor(data: any) {
        if (data) {
            this.boardId = data.boardId;
            this.title = data.title;
            this.timestamp = data.timestamp;
            this.picturePath = data.picturePath;
            this.author = data.author;
            this.authorId = data.autorId;
            this.collaborators = data.collaborators;
            this.lists = data.lists;
        }
    }
}